apps/cmp.c

```diff
@@ -809,7 +809,14 @@ static int load_certs_preliminary(const char *file, STACK_OF(X509) **certs,
     int ret = 0;
 
     if (format == FORMAT_PKCS12) {
-        BIO *bio = bio_open_default(file, 'r', format);
+        FILE *fp = fopen(file, "r");
```

Comment:
`"r"` should be `"rb"`.  `FORMAT_PKCS12` is a binary format, and some platforms (such as Windows) need to know.

However, I fail to understand what was wrong with the `bio_open_default` call.  What's up with this special case?

Reply:
The code there allows failure.  I was hoping @ddvo could answer your question, but he went off and gave detailed feedback on everything else :)

Reply:
I already commented on this above (https://github.com/openssl/openssl/pull/11993#discussion_r435013465):
> Yes, such a change was needed.
The question would have not arisen if the review of #4930 had been finalized and the changes merged, because then the whole load_certs_preliminary() function would be gone.

Reply:
To answer @levitte's new question with a little more detail: 
The change is needed (as long as #4930 is not merged) because `load_certs_autofmt()` may try loading certs via `load_certs_preliminary()` with two different formats, so the app must not terminate if the first try fails.